integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipRoutes.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.bean.eip;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import io.quarkus.arc.Unremovable;
+import org.apache.camel.Consume;
+import org.apache.camel.builder.RouteBuilder;
+
+public class EipRoutes extends RouteBuilder {
+
+    @Override
+    public void configure() {
+        from("direct:dynamicRouter")
+                .dynamicRouter(method("myDynamicRouter", "route"));
+
+    }
+
+    @Produces
+    @Unremovable
```

Comment:
Nit - If a bean is named, then it's unremovable. So the annotation is not strictly required.

Reply:
+1 thanks!